smac/facade/roar_facade.py

```diff
@@ -0,0 +1,130 @@
+from smac.facade.smac_facade import SMAC
+
+__author__ = "Marius Lindauer"
+__copyright__ = "Copyright 2016, ML4AAD"
+__license__ = "3-clause BSD"
+
+
+class ROAR(SMAC):
+
+    def __init__(self,
+                 scenario: Scenario,
+                 tae_runner: ExecuteTARun=None,
+                 runhistory: RunHistory=None,
+                 intensifier: Intensifier=None,
+                 acquisition_function: AbstractAcquisitionFunction=None,
+                 model=None,
+                 runhistory2epm: AbstractRunHistory2EPM=None,
+                 initial_design: InitialDesign=None,
+                 stats: Stats=None,
+                 rng: np.random.RandomState=None):
+        '''
+        Facade to use ROAR default mode
+
+        Parameters
+        ----------
+        scenario: smac.scenario.scenario.Scenario
+            Scenario object
+        tae_runner: ExecuteTARun
+            object that implements the following method to call the target
+            algorithm (or any other arbitrary function):
+            run(self, config)
+            If not set, it will be initialized with the tae.ExecuteTARunOld()
+        runhistory: RunHistory
+            runhistory to store all algorithm runs
+        intensifier: Intensifier
+            intensification object to issue a racing to decide the current incumbent
+        acquisition_function : AcquisitionFunction
+            Object that implements the AbstractAcquisitionFunction. Will use
+            EI if not set.
+        model : object
+            Model that implements train() and predict(). Will use a
+            RandomForest if not set.
+        runhistory2epm : RunHistory2EMP
+            Object that implements the AbstractRunHistory2EPM. If None,
+            will use RunHistory2EPM4Cost if objective is cost or
+            RunHistory2EPM4LogCost if objective is runtime.
+        initial_design: InitialDesign
+            initial sampling design
+        stats: Stats
+            optional stats object
+        rng: np.random.RandomState
+            Random number generator
+        '''
+        self.logger = logging.getLogger("ROAR")
+
+        # initialize random number generator
+        if rng is None:
+            num_run = np.random.randint(1234567980)
+            rng = np.random.RandomState(seed=num_run)
+        elif isinstance(rng, int):
+            num_run = rng
+            rng = np.random.RandomState(seed=rng)
+        elif isinstance(rng, np.random.RandomState):
+            num_run = rng.randint(1234567980)
+            rng = rng
+        else:
+            raise TypeError('Unknown type %s for argument rng. Only accepts '
+                            'None, int or np.random.RandomState' % str(type(rng)))
+
+        # initial EPM
+        if model is None:
+            #use random predictions to simulate random sampling of configurations
+            model = RandomEpm(rng=rng)
+
+        # initial conversion of runhistory into EPM data
```

Comment:
right now, we use no imputer at all, which should be most time-efficient variant, right? We could replace the runhistory2EPM by a random class. However, this function is typically not the bottle neck. 
To save time, we would need to remove the local search and the sorting of the random configuration based on the acq function. However, this would require direct changes in the smbo class which I would like to avoid.